module-gui/gui/widgets/text/core/lines/TextLine.cpp

```diff
@@ -7,148 +7,14 @@ namespace gui
-            shownLetterCount += signsCountToShow;
-            widthUsed += item->widgetArea.w;
-            heightUsed = std::max(heightUsed, item->widgetArea.h);
-            lineContent.emplace_back(item);
-
-            setLineStartConditions(localCursor.getBlockNumber(), localCursor.getPosition());
-
-            localCursor += signsCountToShow;
-
-            if (removeTrailingSpace) {
-                end = TextBlock::End::Newline;
-                return;
-            }
-
-            if (localCursor.checkAndInvalidateBlockChanged() && localCursor.checkPreviousBlockNewLine()) {
-                end = TextBlock::End::Newline;
-                return;
-            }
-
-            // not whole text shown, try again for next line if you want
-            if (signsCountToShow < text.length()) {
-                end = TextBlock::End::None;
-                return;
-            }
-
-        } while (true);
-    }
-
-    unsigned int TextLine::calculateSignsToShow(BlockCursor &localCursor, UTF8 &text, unsigned int space)
-    {
-        auto signsCountToShow = localCursor->getFormat()->getFont()->getCharCountInSpace(text, space);
-
-        // additional one sign to detect potential space as last character in line
-        auto searchSubstring = text.substr(0, signsCountToShow + 1);
-
-        auto newlinePos = searchSubstring.findLast("\n", signsCountToShow);
-        auto spacePos   = searchSubstring.findLast(" ", signsCountToShow);
-
-        // check if space or newline in word detection range
-        if (spacePos <= signsCountToShow - text::word_detection_range) {
-            spacePos = UTF8::npos;
-        }
-
-        if (newlinePos <= signsCountToShow - text::word_detection_range) {
-            newlinePos = UTF8::npos;
-        }
-
-        // check if space found and no newline at end
-        if (spacePos != UTF8::npos && newlinePos == UTF8::npos) {
-            // if line ends on space remove it when drawing
-            if (spacePos >= signsCountToShow) {
-                removeTrailingSpace = true;
-            }
-
-            // add one to include space in the line
-            signsCountToShow = spacePos + 1;
-
-            breakLineDashAddition = false;
-        }
-        // if sings still left in text add dash sign
-        else if (signsCountToShow != 0 && signsCountToShow < text.length()) {
-            // decrease character shown count by one to fit dash
-            signsCountToShow      = signsCountToShow - 1;
-            breakLineDashAddition = true;
-        }
-
-        return signsCountToShow;
-    }
-
-    UTF8 TextLine::textToPrint(unsigned int signsCountToShow, UTF8 &text)
-    {
-        auto textToPrint = text.substr(0, signsCountToShow);
-
-        if (removeTrailingSpace) {
-            textToPrint = text.substr(0, signsCountToShow - 1);
-        }
-
-        if (breakLineDashAddition) {
-            textToPrint = textToPrint + "-";
-        }
-
-        return textToPrint;
-    }
+    {}
```

Comment:
pure virtual? :)

Reply:
Yep -> that can be Pure virtual.

Reply:
I'm pretty sure we can work something the best of both worlds here, saying that - to be discussed :)